opentelemetry-sdk/src/testing/logs/in_memory_exporter.rs

```diff
@@ -148,8 +148,17 @@ impl InMemoryLogExporter {
     /// ```
     ///
     pub fn get_emitted_logs(&self) -> LogResult<Vec<LogDataWithResource>> {
-        let logs_guard = self.logs.lock().map_err(LogError::from)?;
-        let resource_guard = self.resource.lock().map_err(LogError::from)?;
+        let logs_guard = self.logs.lock().map_err(|_| {
```

Comment:
The mutex poisoned errors here are an implementation detail of `InMemoryLogExporter` - that is, they have nothing to do with the common shape of a `LogExporter`. 

So, I removed the automatic `From` for poisoned on the `LogError` type and explicitly map detailed errors through here.